File: 2022/06/solution.rb
#!ruby

input = (STDIN.tty? ? DATA : STDIN).readlines(chomp: true)
input = DATA.readlines(chomp: true) if input.size == 0

PACKET_SIZE = 14

input.
  first.
  chars.
  each_cons(PACKET_SIZE).
  with_index do |cons, index|
    if cons.uniq.size == PACKET_SIZE
      p index + PACKET_SIZE
      exit
    end
  end

__END__
mjqjpqmgbljsphdztnvjfqwrcgsmlb

File: 2022/06/solution.py
#!python

import sys

EXAMPLE = """
mjqjpqmgbljsphdztnvjfqwrcgsmlb
"""

input = EXAMPLE.splitlines()[1:] if sys.stdin.isatty() else sys.stdin.read().splitlines()
if len(input) == 0: input = input = EXAMPLE.splitlines()[1:]

input = input[0]
PACKET_SIZE = 14

cons = [input[i:i+PACKET_SIZE] for i in range(len(input)-PACKET_SIZE+1)]
for i, con in enumerate(cons):
    if (len(set(con)) == PACKET_SIZE):
       print(i + PACKET_SIZE)
       break

